packages/server/src/course/course.controller.ts

```diff
@@ -893,4 +896,84 @@ export class CourseController {
       });
     return;
   }
+
+  // UPDATE course_settings_model SET selectedFeature = false WHERE courseId = selectedCourseId;
+  // will also create a new course settings record if it doesn't exist for the course
+  @Patch(':id/features')
+  @UseGuards(JwtAuthGuard, CourseRolesGuard)
+  @Roles(Role.PROFESSOR)
+  async enableDisableFeature(
+    @Param('id') courseId: number,
+    @Body() body: CourseSettingsRequestBody,
+  ): Promise<void> {
+    // fetch existing course settings
+    let courseSettings = await CourseSettingsModel.findOne({
+      where: { courseId: courseId },
+    });
+
+    // if no course settings exist yet, create new course settings for the course
+    if (!courseSettings) {
+      // first make sure the course exists in course table (this check might not be needed as the guards already make sure the user is in the course (therefore the course must exist), but this is a rare function to be called so the small performance hit is acceptable for later safety)
+      const course = await CourseModel.findOne({
+        where: { id: courseId },
+      });
+      if (!course) {
+        throw new NotFoundException(
```

Comment:
Should we throw exception instead of res.status(HttpStatus.NotFound)?

Reply:
Are they not the same?

Reply:
okay so apparently throwing exceptions is actually a good practice with nestjs, so it should be fine?

From chatGPT:
In the context of a Node.js application using Express.js or similar frameworks, throwing an error like `NotFoundException` will indeed terminate the current request and send an error response to the client. However, it should not crash the entire server process unless it's an unhandled exception.

In NestJS, which is a framework for building efficient, reliable and scalable server-side applications, the `NotFoundException` is a part of the framework's exception layer. Throwing a `NotFoundException` will result in a HTTP response with a 404 status code. NestJS catches the exception and handles it appropriately, so it won't crash the process.

Using `res.status(HttpStatus.NotFound)` is another way to send a 404 response. This is more of an Express.js style of handling errors and it won't crash the server either.

Both approaches are valid, but if you're using NestJS, it's recommended to use its exception layer (i.e., throw a `NotFoundException`) because it provides a consistent way to handle exceptions across your application. It also keeps your controllers' code cleaner and more focused on their primary responsibilities.